www/viewgame

```diff
@@ -640,41 +655,61 @@ if (isset($_REQUEST['ifiction']))
         while ([$id, $ext] = mysql_fetch_row($result)) {
             $fmtMap[$id] = $ext;
         }
-        echo "<downloads><links>";
+        $links_section = [];
+        $ifdb_section['downloads'] = ['links' => &$links_section];
         foreach ($links as $l) {
-            echo "<link>";
-            xmlField("url", $l['url']);
-            if ($l['title']) xmlField("title", $l['title']);
-            if ($l['desc']) xmlField("desc", $l['desc']);
-            if ($l['isgame']) echo "<isGame/>";
-            if ($l['fmtexternid']) xmlField("format", $l['fmtexternid']);
+            $link_obj = [
+                'url' => $l['url'],
+            ];
+            if ($l['title']) $link_obj['title'] = $l['title'];
+            if ($l['desc']) $link_obj['desc'] = $l['desc'];
+            if ($l['isgame']) $link_obj['isGame'] = true;
+            if ($l['fmtexternid']) $link_obj['format'] = $l['fmtexternid'];
             if (isset($l['osext'])) {
                 if (isset($l['osvsn'])) {
-                    xmlField("os", $l['osext'] . "." . $l['osvsnext']);
+                    $link_obj['os'] = $l['osext'] . "." . $l['osvsnext'];
                 } else {
-                    xmlField("os", $l['osext'] . ".");
+                    $link_obj['os'] = $l['osext'] . ".";
                 }
             }
-            if ($l['compression']) xmlField("compression", $fmtMap[$l['compression']]);
-            if ($l['compressedprimary']) xmlField("compressedPrimary", $l['compressedprimary']);
-            echo "</link>";
+            if ($l['compression']) $link_obj['compression'] = $fmtMap[$l['compression']];
+            if ($l['compressedprimary']) $link_obj['compressedPrimary'] = $l['compressedprimary'];
+
+            if ($json) {
+                $links_section[] = $link_obj;
+            } else {
+                $links_section[] = ['link' => $link_obj];
+            }
         }
-        echo "</links></downloads>";
     }
     if (!$should_hide) {
-        xmlField("averageRating", $ratingAvg);
-        list($stars) = roundStars($ratingAvg);
-        xmlField("starRating", $stars);
-        xmlField("ratingCountAvg", $ratingCntAvg);
-        xmlField("ratingCountTot", $ratingCntTot);
+        $ifdb_section['averageRating'] = $ratingAvg;
+        [$stars] = roundStars($ratingAvg);
+        if ($stars) {
+            $ifdb_section['starRating'] = $stars;
+        }
+        $ifdb_section['ratingCountAvg'] = $ratingCntAvg;
+        $ifdb_section['ratingCountTot'] = $ratingCntTot;
     }
-    echo "</ifdb>";
 
-    // main end tags
-    echo "</story>"
-        . "</ifindex>";
+    // Convert object from latin1 to UTF-8
+    array_walk_recursive($story_obj, function(&$value, $key) {
+        if (is_string($value)) {
+            $value = html_entity_decode($value);
+        }
+    });
```

Comment:
As we were discussing earlier, I've merged #1002 and run the admin tool in production (and I filed #1026 to remove special cases for HTML entities).

I think we should remove this `array_walk` code and merge #1026, and then merge this PR.